src/v7/add-site-preview-secret.ts

```diff
@@ -0,0 +1,143 @@
+    updateValuesTplFile();
+}
+
+function updateApiFiles1() {
+    console.log("Add sitePreviewSecret to api/src/config/environment-variables.ts ...");
+
+    const project = new Project({ tsConfigFilePath: "./api/tsconfig.json" });
+
+    const sourceFile = project.getSourceFile("api/src/config/environment-variables.ts");
+    if (!sourceFile) {
+        console.error("  Could not file file, make sure to add SITE_PREVIEW_SECRET");
+        return;
+    }
+    const cls = sourceFile.getClass("EnvironmentVariables");
+    if (!cls) {
+        console.error("  Could not class EnvironmentVariables, make sure to add SITE_PREVIEW_SECRET");
+        return;
+    }
+
+    cls.addMember("\n@IsString()\n@MinLength(16)\nSITE_PREVIEW_SECRET: string;");
+    sourceFile.saveSync();
+
+    console.log("  finished");
+}
+
+async function updateApiFiles2() {
+    console.log("Add sitePreviewSecret to api/src/config/config.ts ...");
+
+    const project = new Project({ tsConfigFilePath: "./api/tsconfig.json" });
+
+    const sourceFile = project.getSourceFile("api/src/config/config.ts");
+    if (!sourceFile) {
+        console.error("  Could not file file, make sure to add sitePreviewSecret: envVars.SITE_PREVIEW_SECRET");
+        return;
+    }
+
+    sourceFile
+        .getFunction("createConfig")
+        ?.getBody()
+        ?.getDescendantsOfKind(SyntaxKind.ReturnStatement)[0]
+        .getChildrenOfKind(SyntaxKind.ObjectLiteralExpression)[0]
+        .addProperty("sitePreviewSecret: envVars.SITE_PREVIEW_SECRET,");
+
+    sourceFile.saveSync();
+
+    console.log("  finished");
+}
+
+async function updateApiFiles3() {
+    console.log("Add sitePreviewSecret to api/src/app.module.ts ...");
+
+    const project = new Project({ tsConfigFilePath: "./api/tsconfig.json" });
+
+    const sourceFile = project.getSourceFile("api/src/app.module.ts");
+    if (!sourceFile) {
+        console.error("  Could not file file, make sure to add sitePreviewSecret to PageTreeModule");
+        return;
+    }
+
+    sourceFile
+        .getClassOrThrow("AppModule")
+        .getMethodOrThrow("forRoot")
+        .getBody()
+        ?.getDescendantsOfKind(SyntaxKind.CallExpression)
+        .find((call) => call.getText().includes("PageTreeModule.forRoot"))
+        ?.getChildrenOfKind(SyntaxKind.ObjectLiteralExpression)[0]
+        .addProperty("sitePreviewSecret: config.sitePreviewSecret,");
+
+    sourceFile.saveSync();
+
+    console.log("  finished");
+}
+
+function updateDotEnvFile() {
+    console.log("Update .env");
+    if (!fs.existsSync(".env")) {
+        console.error("  could not find file, please make sure to add SITE_PREVIEW_SECRET manually");
+        return;
+    }
+    fs.appendFile(".env", `\nSITE_PREVIEW_SECRET=${crypto.randomBytes(8).toString("hex")}\n`, (err) => {
+        console.error(`  ${err}`);
+    });
+    console.log("  finished");
+}
+
+function updateValuesTplFile() {
+    const valuesFileName = "deployment/helm/values.tpl.yaml";
```

Comment:
This won't work for lots of our projects that have been started pre Comet 7. However, this probably won't work anyway, since the deployment needs to be adapted for Comet 7 anyway. Is this okay for us?